File: 2024_01/Exercicios/C#/palindromo.cs
using System;

class Program{
    static bool Palindromo(String str){
        int i = 0;
        int j = str.Length - 1;

        while(i < j){

            if (str[i] != str[j]){
                return false;
            }

            i++;
            j--;
        }

        return true;
    }

    static void Main(string[] args){
        string line;

        while((line = Console.ReadLine()) != "FIM"){
            if(Palindromo(line)){
                Console.WriteLine("SIM");
            } else{
                Console.WriteLine("NAO");
            }
        }
    }
}

File: 2024_01/Exercicios/java/palindromo.java
import java.util.Scanner;

public class palindromo{
    public static boolean Palindromo(String str){
        int i = 0;
        int j = str.length() - 1;

        while(i < j){
            if(str.charAt(i) != str.charAt(j)){
                return false;
            } 
            i++;
            j--;
        }
        return true;
    }

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        String line;

        while (!(line = sc.nextLine()).equals("FIM")) {
            if (Palindromo(line)) {
                System.out.println("SIM");
            } else {
                System.out.println("NAO");
            }
        }

        sc.close();
    }
}
